src/sdk/cpp/adapters/mqtt/src/mqtt_adapter.cpp

```diff
@@ -137,41 +158,42 @@ void MqttClient::Run() {
     }
 }
 
-void MqttClient::Stop() { ChangeState(State::kStopping); }
-
-void MqttClient::Subscribe(const std::string& topic, const int qos) {
-    log::Debug() << "Subscribing to " << topic << " qos=" << qos;
-    topics_.push_back(std::make_pair(topic, qos));
-    //client_.subscribe(topic, qos);
+void MqttProtocolAdapter::Stop() {
+    ChangeState(State::kStopping);
 }
 
-void MqttClient::ChangeState(State state) {
-    std::lock_guard<std::mutex> lock(state_mutex_);
-    if (next_state_ != state) {
-        switch (state) {
-            case State::kDisconnected:
-                log::Debug() << "Changing state to: [Disconnected].";
-                break;
-            case State::kConnecting:
-                log::Debug() << "Changing state to: [Connecting].";
-                break;
-            case State::kConnected:
-                log::Debug() << "Changing state to: [Connected].";
-                break;
-            case State::kStopping: {
-                log::Debug() << "Changing state to: [Stopping].";
-            } break;
-        }
-        next_state_ = state;
-    }
-}
+void MqttProtocolAdapter::Publish(const std::string& topic, const std::string& data, const PublishOptions&) {
+    auto full_topic = topic_ns_ + topic;
 
-void MqttClient::Publish(const std::string& topic, const std::string& data) {
     log::Debug() << "Publishing message.";
-    log::Debug() << "\ttopic: '" << topic << "'";
+    log::Debug() << "\ttopic: '" << full_topic << "'";
```

Comment:
I think the logs (all) have to be merged with the previous PR (https://github.com/GENIVI/iot-event-analytics/pull/79)